test/integration/overload_integration_test.cc

```diff
@@ -213,16 +223,98 @@ TEST_P(OverloadIntegrationTest, StopAcceptingConnectionsWhenOverloaded) {
   EXPECT_FALSE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_,
                                                          std::chrono::milliseconds(1000)));
 
-  // Reduce load a little to allow the connection to be accepted but then immediately reject the
-  // request.
+  // Reduce load a little to allow the connection to be accepted.
   updateResource(0.9);
   test_server_->waitForGaugeEq("overload.envoy.overload_actions.stop_accepting_connections.active",
                                0);
+  EXPECT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
+  EXPECT_TRUE(fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_));
+  ASSERT_TRUE(upstream_request_->waitForHeadersComplete());
+  ASSERT_TRUE(upstream_request_->waitForData(*dispatcher_, 10));
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "202"}}, true);
```

Comment:
Seems like the meaning of this test changed, the request is now getting to the upstream.

Reply:
That was intentional. This test was checking two overload behaviors at once: first that connections would not be accepted, then that when they were, new requests would be rejected. This seemed like a good way to disentangle those two.